src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -434,17 +540,30 @@ private static void checkSorted(final File inputFile, final Feature lastFeature,
          * @param codec
          */
         public FeatureIterator(final File inputFile, final FeatureCodec<FEATURE_TYPE, SOURCE> codec) {
-            if (inputFile == null) {
+            this(ValidationUtils.nonNull(inputFile, "FeatureIterator input file cannot be null").toPath(), codec);
+        }
+
+        /**
+         *
+         * @param inputPath The path from which to read. Stream for reading is opened on construction. May not be null.
+         * @param codec
+         */
+        public FeatureIterator(final Path inputPath, final FeatureCodec<FEATURE_TYPE, SOURCE> codec) {
+            if (inputPath == null) {
                 throw new IllegalArgumentException("FeatureIterator input file cannot be null");
             }
             this.codec = codec;
-            this.inputFile = inputFile;
+
+            // We _MUST_ call getPathToDataFile here to preserve behavior of codecs that rely on config files
+            // that point to other files to be indexed. This is CRITICALLY IMPORTANT to downstream tools.
+            this.inputPath = Paths.get(codec.getPathToDataFile(inputPath.toUri().toString()));
```

Comment:
ugh...  path -> uri -> string -> string -> path   Not much to do about it though :(

Reply:
Yeah, I considered changing `getPathToDataFile` to return a `Path` instead of a String, but decided that was too much scope creep.